06week/main.js

```diff
@@ -0,0 +1,283 @@
+//const assert = require("assert");
+
+// defines constant array of person objects
+const arrOfPeople = [
+  {
+    id: 2,
+    name: "Charles Young",
+    age: 55,
+    skillSet: "welding",
+    placeBorn: "Omaha, Nebraska"
+  },
+  {
+    id: 3,
+    name: "Judy Twilight",
+    age: 35,
+    skillSet: "fishing",
+    placeBorn: "Louisville, Kentucky"
+  },
+  {
+    id: 4,
+    name: "Cynthia Doolittle",
+    age: 20,
+    skillSet: "tic tac toe",
+    placeBorn: "Pawnee, Texas"
+  },
+  {
+    id: 5,
+    name: "John Willouby",
+    age: 28,
+    skillSet: "pipe fitting",
+    placeBorn: "New York, New York"
+  },
+  {
+    id: 6,
+    name: "Stan Honest",
+    age: 20,
+    skillSet: "boom-a-rang throwing",
+    placeBorn: "Perth, Australia"
+  },
+  {
+    id: 7,
+    name: "Mia Watu",
+    age: 17,
+    skillSet: "acrobatics",
+    placeBorn: "Los Angeles, California"
+  },
+  {
+    id: 8,
+    name: "Walter Cole",
+    age: 32,
+    skillSet: "jump rope",
+    placeBorn: "New Orleans, Louisiana"
+  }
+];
+
+// empty arrays waiting to be filled
+const listOfPlayers = [];
+const blueTeam = [];
+const redTeam = [];
+
+// constructs the player object
+class player {
+  // constructor takes the following args
+  constructor(
+    person,
+    canThrowBall,
+    canDodgeBall,
+    hasPaid,
+    isHealthy,
+    yearsExperience
+  ) {
+    // for loop that loops through the pre-existing key value pairs and adds them as attributes
+    // to the resulting player object
+    for (var [key, value] of Object.entries(person)) {
+      console.log("key", key, "value", value);
+      this[key] = value;
+    }
+    // sets additional attributes on the resulting player object
+    this.canThrowBall = canThrowBall;
+    this.canDodgeBall = canDodgeBall;
+    this.hasPaid = hasPaid;
+    this.isHealthy = isHealthy;
+    this.yearsExperience = yearsExperience;
+  }
+}
+
+// constructs the blueTeammate object
+class blueTeammate {
```

Comment:
I would have expected a single class  Teammate, that had a color/team as an attribute